src/GameplaySystem.cpp

```diff
@@ -11,11 +12,46 @@ GameplaySystem::GameplaySystem() {
 
 void GameplaySystem::update() {
     // Update game state here
+    EventSystem::queue_event (
+        Event (
+            EventType::EXAMPLE_SHIP_IDLE_EVENT,
```

Comment:
IDLE is kind of an oxymoron to this, can we change it to `ROTATE_GAMEOBJECT`, I think this name matches the params below much better.

Reply:
No. I do not want to have to queue an event for each change in the object. The ship could be translating forward and back during it's idle state, rotation is just an example.

Also, this event is just an example of how the GameplaySystem might interact with other systems to manipulate game objects. This is not an actual event that will be carried into the game.